hack/golangci.yaml.in

```diff
@@ -11,258 +11,279 @@
-
-    # https://github.com/kubernetes/kubernetes/issues/117288#issuecomment-1507031224
-    - linters:
-        - stylecheck
-      text: "ST1023: should omit type .* from declaration; it will be inferred from the right-hand side"
-    {{- end}}
-
 linters:
-  disable-all: {{if .Base -}} true {{- else -}} false {{- end}}
+  default: {{if .Base -}} none {{- else -}} standard {{- end}}
   enable: # please keep this alphabetized
     - depguard
+    {{- if .Hints}}
+    - errorlint
+    {{- end}}
     - forbidigo
     - ginkgolinter
     - gocritic
     - govet
-    {{- if .Hints}}
-    - errorlint
-    {{- end}}
-    - ineffassign
     - logcheck
     - revive
     - staticcheck
-    - stylecheck
     - testifylint
     - unused
   {{- if not .Base }}
     - usestdlibvars
-  {{- end}}
-
-linters-settings: # please keep this alphabetized
-  custom:
-    logcheck:
-      # Installed there by hack/verify-golangci-lint.sh.
-      path: ../_output/local/bin/logcheck.so
-      description: structured logging checker
-      original-url: k8s.io/logtools/logcheck
-      settings:
-        config: |
-          {{include "hack/logcheck.conf" | indent 10 | trim}}
-  depguard:
-    rules:
-      main:
-        files:
-          - $all
-          - "!$test"
-          - "!**/test/**"
-        deny:
-          - pkg: "github.com/google/go-cmp/cmp"
-            desc: "cmp is allowed only in test files"
-  forbidigo:
-    analyze-types: true
-    forbid:
-    - p: ^managedfields\.ExtractInto$
-      pkg: ^k8s\.io/apimachinery/pkg/util/managedfields$
-      msg: should not be used because managedFields was removed
-    - p: \.Extract
-      pkg: ^k8s\.io/client-go/applyconfigurations/
-      msg: should not be used because managedFields was removed
-    - p: \.Add$
-      pkg: ^k8s\.io/component-base/featuregate$
-      msg: should not use Add, use AddVersioned instead
-    {{- if .Hints}}
-    - p: ^gomega\.BeTrue$
-      pkg: ^github.com/onsi/gomega$
-      msg: "it does not produce a good failure message - use BeTrueBecause with an explicit printf-style failure message instead, or plain Go: if ... { ginkgo.Fail(...) }"
-    - p: ^gomega\.BeFalse$
-      pkg: ^github.com/onsi/gomega$
-      msg: "it does not produce a good failure message - use BeFalseBecause with an explicit printf-style failure message instead, or plain Go: if ... { ginkgo.Fail(...) }"
-    {{- end}}
-  {{- if .Base }}
-  gocritic:
-    enabled-checks:
-      - equalFold
-      - boolExprSimplify
-  {{- end}}
-  revive:
-    # Only these rules are enabled.
+  {{- end }}
+
+  exclusions:
+    # The default excludes disable the "should have comment or be unexported" check from revive.
+    # We want that to be enabled, therefore we have to disable all default excludes and
+    # add those back one-by-one that we want. See https://github.com/golangci/golangci-lint/issues/456#issuecomment-617470264
+    paths:
+      - third_party
```

Comment:
Does this comment here make sense? Does it still apply anywhere at all?

Reply:
With `third_party` excluded here, do we need the `formatters.exclusions.path` entry above?

Why is `vendor` handled differently?

Reply:
The exclusion paths should be "duplicated" inside the  `formatters` and `linters` sections because this is related to 2 different commands.

`vendor` is ignored by the Go tooling for linters, and by an internal rule for formatters.